pkgs/development/libraries/pulsar-client-cpp/default.nix

```diff
@@ -0,0 +1,80 @@
+{ stdenv
+, lib
+, pkgs
+, fetchurl
+, version ? "2.8.1"
+, sha512 ? "23i95k6gs5lh6gw6v6hybv6z8rfzc3mfgj2agb6iicad9ia2sf46wr3rqzi9bdhp0c990vcvn72bkz13i0hnh1hj5gnwqin339a2g0x"
+, boost17x
+, clang
+, llvmPackages
+, cmake
+, curl
+, gcc
+, jsoncpp
+, log4cxx
+, openssl
+, pkgconfig
+, protobuf
+, python37
+, snappy
+, zlib
+, zstd}:
+
+let
+  # Not really sure why I need to do this.. If I call clang-tools without the override it defaults to clang_10
+  clang-tools = pkgs.clang-tools.override {inherit stdenv llvmPackages;};
+in
+  stdenv.mkDerivation rec {
+    pname = "pulsar-client-cpp";
+    inherit version;
+
+    src = fetchurl {
+      inherit sha512;
+      url = "https://archive.apache.org/dist/pulsar/pulsar-${version}/apache-pulsar-${version}-src.tar.gz";
+    };
+
+    sourceRoot = "apache-pulsar-${version}-src/pulsar-client-cpp";
+
+    # python37 used in cmake script to calculate some values
+    # clang-tools needed for clang-format etc
```

Comment:
Do we really need clang-format? Sounds like they are used in some sort of lint CI which we can happily skip.

Can we use python39 through python3? If so please patch the python37 out so that we don't need to worry about that.